src/main/scala/it/pagopa/interop/backendforfrontend/error/BFFErrors.scala

```diff
@@ -89,4 +89,7 @@ object BFFErrors {
   final case class PurposeVersionDraftNotFound(purposeId: UUID)
       extends ComponentError("0019", s"Version in DRAFT state for Purpose $purposeId not found")
 
+  final case class AgreementNotFound(requesterId: UUID)
+      extends ComponentError("0020", s"Agreement of requester $requesterId not found")
```

Comment:
This seems to be always used for the consumer
```suggestion
  final case class AgreementNotFound(consumerId: UUID)
      extends ComponentError("0020", s"Agreement of consumer $consumerId not found")
```